Ginger/GingerCore/ALM/OctaneCore.cs

```diff
@@ -226,5 +365,536 @@ public override bool IsServerConnected()
+                                    paramSelectedValue = paramSelectedValue.Substring(3);//get value without "$$_"
+                                }
+                            }
+                            else if (paramSelectedValue != "<Empty>")
+                            {
+                                isflowControlParam = true;
+                            }
+
+                            //check if already exist param with that name
+                            VariableBase stepActivityVar = stepActivity.Variables.Where(x => x.Name.ToUpper() == param.ToUpper()).FirstOrDefault();
+                            if (stepActivityVar == null)
+                            {
+                                //#Param not exist so add it
+                                if (isflowControlParam.Value)
+                                {
+                                    //add it as selection list param                               
+                                    stepActivityVar = new VariableSelectionList();
+                                    stepActivityVar.Name = param;
+                                    stepActivity.AddVariable(stepActivityVar);
+                                    stepActivity.AutomationStatus = eActivityAutomationStatus.Development;//reset status because new flow control param was added
+                                }
+                                else
+                                {
+                                    //add as String param
+                                    stepActivityVar = new VariableString();
+                                    stepActivityVar.Name = param;
+                                    ((VariableString)stepActivityVar).InitialStringValue = paramSelectedValue;
+                                    stepActivity.AddVariable(stepActivityVar);
+                                }
+                            }
+                            else
+                            {
+                                //#param exist
+                                if (isflowControlParam.Value)
+                                {
+                                    if (!(stepActivityVar is VariableSelectionList))
+                                    {
+                                        //flow control param must be Selection List so transform it
+                                        stepActivity.Variables.Remove(stepActivityVar);
+                                        stepActivityVar = new VariableSelectionList();
+                                        stepActivityVar.Name = param;
+                                        stepActivity.AddVariable(stepActivityVar);
+                                        stepActivity.AutomationStatus = eActivityAutomationStatus.Development;//reset status because flow control param was added
+                                    }
+                                }
+                                else if (isflowControlParam.Value)
```

Comment:
Codacy found an issue: [This branch duplicates the one on line 739.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6317725)